src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,489 @@
+                    y = y + dv
+
+                    # If it is now outside domain, then save age and dont process further with outside lam flag
+                    # Support cyclic domains like K-SCALE, where x coord out of domain gets moved through dateline.
+                    if cyclic:
+                        if (
+                            x <= -1
+                        ):  # as for example -0.3 would still be in domain, but x_arr.shape-0.3 would result in index error
+                            x = x_arr.shape[3] + x  # wrap back around dateline
+                        elif x >= x_arr.shape[3]:
+                            x = x_arr.shape[3] - x
+                    else:
+                        if x < 0 or x >= x_arr.shape[3]:
+                            ageofair_local[leadtime, lat_pnt] = n * dt
+                            outside_lam = True
+
+                    if y < 0 or y >= x_arr.shape[2]:
+                        ageofair_local[leadtime, lat_pnt] = n * dt
+                        outside_lam = True
+
+    # Save 3d array containing age of air
+    np.save(tmpdir + "/aoa_frag_" + str(lon_pnt).zfill(4) + ".npy", ageofair_local)
+
+
+def compute_ageofair(
+    XWIND: Cube,
+    YWIND: Cube,
+    WWIND: Cube,
+    GEOPOT: Cube,
+    plev: int,
+    incW: bool,
+    cyclic: bool = False,
+    multicore=True,
+):
+    """
+    Compute back trajectories for a given forecast.
+
+    This allows us to determine when air entered through the boundaries. This will run on all available
+    lead-times, and thus return an age of air cube of shape ntime, nlat, nlon. It supports multiprocesing,
+    by iterating over longitude, or if set as None, will run on a single core, which is easier for debugging.
+    This function supports ensembles, where it will check if realization dimension exists and if so, loop
+    over this axis.
+
+    Arguments
+    ----------
+    XWIND: Cube
+        An iris cube containing the x component of wind on pressure levels, on a 0p5 degree grid.
+        Requires 4 dimensions, ordered time, pressure, latitude and longitude. Must contain at
+        least 2 time points to compute back trajectory.
+    YWIND: Cube
+        An iris cube containing the y component of wind on pressure levels, on a 0p5 degree grid.
+        Requires 4 dimensions, ordered time, pressure, latitude and longitude. Must contain at
+        least 2 time points to compute back trajectory.
+    WWIND: Cube
+        An iris cube containing the w component of wind on pressure levels, on a 0p5 degree grid.
+        Requires 4 dimensions, ordered time, pressure, latitude and longitude. Must contain at
+        least 2 time points to compute back trajectory.
+    GEOPOT: Cube
+        An iris cube containing geopotential height on pressure levels, on a 0p5 degree grid.
+        Requires 4 dimensions, ordered time, pressure, latitude and longitude. Must contain at
+        least 2 time points to compute back trajectory.
```

Comment:
Given the requirements are all the same, it might be clearer to omit them for all but the first cube, and just say that it must have the same form as the XWIND.